.github/workflows/pages-ci.yml

```diff
@@ -21,13 +21,15 @@ jobs:
           node-version: ${{ matrix.node_version }}
 
       - uses: actions/cache@v2
+        id: cache
         with:
-          path: ~/.npm
+          path: |
+            node_modules
+            */*/node_modules
           key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-node-
 
-      - name: Install NPM dependencies
+      - name: Install NPM Dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
```

Comment:
I don't think this statement is right, we want it to run every time, npm install also sets up the workspaces symlinks etc. 
```suggestion
```

Reply:
I think I agree with @threepointone here; when there is no change in the `package-lock.json` then the `npm install` will be super quick.

Reply:
Not disagreeing, but what would the workspace symlinks affect in the CI/CD that would make it necessary to still do the `npm install`

Reply:
> I think I agree with @threepointone here; when there is no change in the `package-lock.json` then the `npm install` will be super quick.

True, it is only a couple of seconds more.

Reply:
Right now we don't do anything that imports something from workspace into another, but that'll change soon I think. For example, we call `npx wrangler dev` from inside `example-worker-app` or `create-workers-app`. Or when wrangler has an api, we'll call `import api from 'wrangler/api'` or something from a sibling workspace. Those will fail in CI, and we won't know why unless we investigate the github workflows. 
Another situation - we update `package.json` but not update `package-lock.json` (for example, when we merge a release). That'll cause stuff to subtly break, and it won't be clear why. (In a future PR, we should probably call `npm install` and fail if any files, probably package-lock.json, has any uncommitted changes. The release workflow should probably also call `npm install` after merging the PR.)

Reply:
Thanks!

Reply:
Yarn has `--frozen-lockfile` which will fail if the lock file needs updating given changes to `package.json`. But I don't think npm has such a thing?

Reply:
probably `npm ci` except I think that blows away `node_modules`

Reply:
Ahah! Yes. From https://docs.npmjs.com/cli/v8/commands/npm-ci

> - If dependencies in the package lock do not match those in package.json, npm ci will exit with an error, instead of updating the package lock.
> - If a node_modules is already present, it will be automatically removed before npm ci begins its install.

What I have seen in the past, though, is to cache the "npm cache" rather than node_modules itself. This is probably why the original workflow was caching `~/.npm`, which is where the cache is stored. This means that there is no need to download packages, and I think if we then use `npm ci` then the installation would be pretty fast since `npm ci` does not need to compute dependencies.